github_connector/tests/test_github_analysis_rule.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (C) 2020: Odoo Community Association (OCA)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+import logging
+
+from odoo.tests.common import SavepointCase
+
+_logger = logging.getLogger(__name__)
+
+
+class TestGithubAnalysisRule(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestGithubAnalysisRule, cls).setUpClass()
+
+        cls.analysis_rule_group_misc = cls.env["github.analysis.rule.group"].create(
+            {"name": "Misc"}
+        )
+        cls.analysis_rule_python = cls.env["github.analysis.rule"].create(
+            {
+                "name": "Python files",
+                "group_id": cls.analysis_rule_group_misc.id,
+                "paths": "/*/*.py",
+            }
+        )
+        cls.analysis_rule_xml = cls.env["github.analysis.rule"].create(
+            {
+                "name": "Xml files",
+                "group_id": cls.analysis_rule_group_misc.id,
+                "paths": "/*/*.xml",
+            }
+        )
+        cls.analysis_rule_test = cls.env["github.analysis.rule"].create(
+            {
+                "name": "Test files",
+                "group_id": cls.analysis_rule_group_misc.id,
+                "paths": "/tests/*.py",
+            }
+        )
+        cls.analysis_rule_ocb = cls.env["github.analysis.rule"].create(
+            {
+                "name": "OCB files (.py + .xml)",
+                "group_id": cls.analysis_rule_group_misc.id,
+                "paths": """
+                *.py
+                *.xml
+                !/.*/
+                """,
+            }
+        )
+        cls.organization_oca = cls.env["github.organization"].create(
+            {
+                "name": "Odoo Community Association",
+                "description": "The GitHub repos for all Open Source work around Odoo",
+                "website_url": "https://odoo-community.org/",
+                "github_url": "https://github.com/OCA",
+                "github_login": "OCA",
+            }
+        )
+        cls.env["github.organization.analysis.rule"].create(
+            {
+                "organization_id": cls.organization_oca.id,
+                "analysis_rule_id": cls.analysis_rule_python.id,
+            }
+        )
+        cls.env["github.organization.analysis.rule"].create(
+            {
+                "organization_id": cls.organization_oca.id,
+                "analysis_rule_id": cls.analysis_rule_test.id,
+            }
+        )
+        cls.env["github.organization.analysis.rule"].create(
+            {
+                "organization_id": cls.organization_oca.id,
+                "analysis_rule_id": cls.analysis_rule_xml.id,
+            }
+        )
```

Comment:
Can use multi-create
```suggestion
        cls.env["github.organization.analysis.rule"].create((
            {
                "organization_id": cls.organization_oca.id,
                "analysis_rule_id": cls.analysis_rule_python.id,
            },
            {
                "organization_id": cls.organization_oca.id,
                "analysis_rule_id": cls.analysis_rule_test.id,
            },
            {
                "organization_id": cls.organization_oca.id,
                "analysis_rule_id": cls.analysis_rule_xml.id,
            }
        ))
```